packages/app/src/cli/services/dev/extension.ts

```diff
@@ -109,32 +111,64 @@ export interface ExtensionDevOptions {
 }
 
 export async function devUIExtensions(options: ExtensionDevOptions): Promise<void> {
-  const payloadStoreOptions = {
+  const payloadOptions = {
     ...options,
     websocketURL: getWebSocketUrl(options.url),
   }
-  const bundlePath = options.appWatcher.buildOutputPath
-  const payloadStoreRawPayload = await getExtensionsPayloadStoreRawPayload(payloadStoreOptions, bundlePath)
-  const payloadStore = new ExtensionsPayloadStore(payloadStoreRawPayload, payloadStoreOptions)
 
-  outputDebug(`Setting up the UI extensions HTTP server...`, options.stdout)
-  const httpServer = setupHTTPServer({devOptions: options, payloadStore})
+  // NOTE: Always use `payloadOptions`, never `options` directly. This way we can mutate `payloadOptions` without
+  // affecting the original `options` object and we only need to care about `payloadOptions` in this function.
```

Comment:
Note that we could make `options` of type `Readonly<ExtensionDevOptions>` but there are also downsides as `Readonly` is a somewhat weak guarantee and might make us think code is safe when it really isn't. Not a mandatory change, but something to consider.